lib/ace/edit_session/bracket_match.js

```diff
@@ -37,17 +37,76 @@ var Range = require("../range").Range;
 
 function BracketMatch() {
 
+
+    this.findSurroundingBrackets = function(position) {
+        if ((position.column == 0) && (position.row)) return null;
+
+        var allBrackets = { 
+            left : [this.$findOpeningBracket("}", position, /(\.?.paren)+/), 
+                    this.$findOpeningBracket(']', position, /(\.?.paren)+/), 
+                    this.$findOpeningBracket(')', position, /(\.?.paren)+/)],
+            right : [this.$findClosingBracket("{", position, /(\.?.paren)+/),
+                     this.$findClosingBracket('[', position, /(\.?.paren)+/),
+                     this.$findClosingBracket('(', position, /(\.?.paren)+/)]
+        };
+
+        var left = {row: null, column: null};
+        var right = {row: null, column: null};
+
+        var leftRowArr = [],
+            leftColArr = [],
+            rightColArr = [],
+            rightRowArr = [];
+
+        var length = Math.max(allBrackets.left.length, allBrackets.right.length)
```

Comment:
do you plan to ever have these arrays of different length? It is you who creates the array, it is not user input or something which is no under your control. If you don't trust to yourself so much, I recommend throwing error if lengths are different.